worker/uniter/uniter.go

```diff
@@ -521,34 +522,34 @@ func (u *Uniter) loop(unitTag names.UnitTag) (err error) {
 
 			err = u.translateResolverErr(err)
 
-			switch cause := errors.Cause(err); cause {
-			case nil:
+			switch {
+			case err == nil:
 				// Loop back around.
-			case resolver.ErrLoopAborted:
+			case errors.Is(err, resolver.ErrLoopAborted):
 				err = u.catacomb.ErrDying()
-			case operation.ErrNeedsReboot:
+			case errors.Is(err, operation.ErrNeedsReboot):
 				err = jworker.ErrRebootMachine
-			case operation.ErrHookFailed:
+			case errors.Is(err, operation.ErrHookFailed):
 				// Loop back around. The resolver can tell that it is in
 				// an error state by inspecting the operation state.
 				err = nil
-			case runner.ErrTerminated:
+			case errors.Is(err, runner.ErrTerminated):
 				localState.HookWasShutdown = true
 				err = nil
-			case resolver.ErrTerminate:
+			case errors.Is(err, resolver.ErrUnitDead):
 				err = u.terminate()
-			case resolver.ErrRestart:
+			case errors.Is(err, resolver.ErrRestart):
 				// make sure we update the two values used above in
 				// creating LocalState.
 				charmURL = localState.CharmURL
 				charmModifiedVersion = localState.CharmModifiedVersion
 				// leave err assigned, causing loop to break
-			case jworker.ErrTerminateAgent:
+			case errors.Is(err, jworker.ErrTerminateAgent):
 				// terminate agent
 			default:
 				// We need to set conflicted from here, because error
 				// handling is outside of the resolver's control.
-				if operation.IsDeployConflictError(cause) {
+				if _, is := errors.AsType[*operation.DeployConflictError](err); is {
```

Comment:
hate this syntax, could we keep the helper method and hide it behind that, much nicer to read and much easier to type (IDE will auto complete the method name etc)

Reply:
I disagree but I am happy to take a poll. @hpidcock